src/include/executor/index_scan_executor.h

```diff
@@ -88,6 +105,8 @@ class IndexScanExecutor : public AbstractScanExecutor {
   std::vector<expression::AbstractExpression *> runtime_keys_;
 
   bool key_ready_ = false;
+
+  planner::IndexScanPlan *plan_node_;
```

Comment:
We have discussed this problem. Since node_ is const, we cannot update plan node using node_. That's why I add plan_node_.